### PR TITLE
feat: settings page for CLI Tools

### PR DESCRIPTION
**Pull request by dgolovin**
### What does this PR do?

Adds `Settings/CLI Tools` page to show cli tools registered by extensions using cli API

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?

related to #3781
depends on #4197

### CODE REVIEW

**Comment by benoitf** [comment]
@dgolovin I think the depends on in the description is not the expected one 

replacing
- [ ] https://github.com/containers/podman-desktop/pull/4194 

by 
- [ ] : https://github.com/containers/podman-desktop/pull/4197

**Comment by dgolovin** [comment]
You are correct. Updated.

**Comment by cdrage** [comment]
Fixed the rebase so I can test it 👍

**Review by cdrage — COMMENTED** [review]
Was able to test this against https://github.com/containers/podman-desktop/pull/4388 and it worked really well! The code LGTM besides the missing 1 test.

**Review by cdrage — APPROVED** [review]
Pushed a rebase to your branch @dgolovin and marked this as LGTM!

Once that one test is added it's good on my side 💯

File: packages/renderer/src/lib/preferences/PreferencesCliToolsRendering.svelte [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                {/if}
+              {/if}
+              <span class="my-auto text-gray-400 ml-3 break-words">{cliTool.name}</span>
+            </div>
+          </div>
+        </div>
+        <!-- cli-tools columns -->
+        <div class="grow flex-column divide-gray-900 ml-2">
+          <span class="my-auto text-gray-400 ml-3 break-words">{cliTool.displayName}</span>
+          <div class="float-right text-gray-900 px-2 text-sm">Registered by {cliTool.extensionInfo.label}</div>
```
Comment (cdrage) [inline packages/renderer/src/lib/preferences/PreferencesCliToolsRendering.svelte]: Smallll nitpick, can we have it similar to the mockup with the underlining and saying "Provided by X"?
Screenshot 2023-10-19 at 9 48 08 PM

The full mockup is located here: https://github.com/containers/podman-desktop/issues/3707#issuecomment-1770922558

**Review by cdrage — CHANGES_REQUESTED** [review]

File: packages/renderer/src/lib/preferences/PreferencesPage.svelte [hunk 2]
```diff
@@ -72,6 +73,9 @@ onMount(async () => {
   <Route path="/authentication-providers" breadcrumb="Authentication">
     <PreferencesAuthenticationProvidersRendering />
   </Route>
+  <Route path="/cli-tools" breadcrumb="Authentication">
```
Comment (lstocchi) [inline packages/renderer/src/lib/preferences/PreferencesPage.svelte]: should `breadcrumb` be different?

File: packages/renderer/src/stores/cli-tools.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import type { CliToolInfo } from '../../../main/src/plugin/api/cli-tool-info';
+
+const windowEvents: string[] = ['extensions-started', 'cli-tool-create', 'cli-tool-remove', 'cli-tool-change'];
+const windowListeners = ['system-ready', 'extensions-already-started'];
+
+let extensionsStarted = false;
+
+export async function checkForUpdate(eventName: string): Promise<boolean> {
+  // trigger update after all extensions started
+  if (eventName === 'extensions-started' || eventName === 'extensions-already-started') {
```
Comment (lstocchi) [inline packages/renderer/src/stores/cli-tools.ts]: Should this just be `if ('eventName === 'extensions-already-started')` ? The `extensions-already-started` is triggered after the extensions-started so it looks to me a repetition

File: packages/renderer/src/stores/cli-tools.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { writable, type Writable } from 'svelte/store';
+import { EventStore } from './event-store';
+import type { CliToolInfo } from '../../../main/src/plugin/api/cli-tool-info';
+
+const windowEvents: string[] = ['extensions-started', 'cli-tool-create', 'cli-tool-remove', 'cli-tool-change'];
```
Comment (lstocchi) [inline packages/renderer/src/stores/cli-tools.ts:23]: Should this contain `extension-stopped` so you refresh the store if an extension is removed?

File: packages/renderer/src/lib/preferences/PreferencesCliToolsRendering.svelte (hunk 1 above)
Comment (dgolovin) [reply packages/renderer/src/lib/preferences/PreferencesCliToolsRendering.svelte]: This is changed to 'Registred' as @benoitf suggested in original pull request for CLI API.

File: packages/renderer/src/stores/cli-tools.ts (hunk 3 above)
Comment (dgolovin) [reply packages/renderer/src/stores/cli-tools.ts]: You are right, Loader.svelte sends this event when main window loads.

File: packages/renderer/src/stores/cli-tools.ts (hunk 4 above)
Comment (dgolovin) [reply packages/renderer/src/stores/cli-tools.ts:23]: It is not required, because cli registration is going to be disposed before that.

**Review by lstocchi — APPROVED** [review]
LGTM

**Review by cdrage — APPROVED** [review]

**Pull request merged**